org.eclipse.draw2d/src/org/eclipse/draw2d/DeferredUpdateManager.java

```diff
@@ -266,15 +266,15 @@ protected void releaseGraphics(Graphics graphics) {
 	 * regions.
 	 */
 	protected void repairDamage() {
-		Iterator keys = dirtyRegions.keySet().iterator();
+		Iterator<IFigure> keys = dirtyRegions.keySet().iterator();
 		Rectangle contribution;
 		IFigure figure;
 		IFigure walker;
 
 		while (keys.hasNext()) {
```

Comment:
as we need both key and value wouldn't it be better if we directly iterate over the entry set? i.e. dirtyRegions.entrySet().forEach( ...

Reply:
I'm always hesitant to change more that what's absolutely necessary. But in principle, you're right. A lot of the iterators can be replaced with the for-each paradigm. You can make a similar argument for e.g. the changes to the Thumbnail class.
I'll go over the changes and see if I can simplify it a little bit more.

Reply:
@azoitl Both the DeferredUpdateManager and the Thumbnail also use for-each loops now.

Reply:
The main reason I suggested the change is that I found it strange from a performance reason to iterate over the keys and then do a look up. But now with the forEach loop it looks so much clearer. Thx for the change. Again tested with GEF Classic Logic Editor, 4diac IDE and compiled GMF runtime. Looks good now.